### PR TITLE
Move metadata from `File` to `EventAttachment`

### PR DESCRIPTION
**Pull request by Swatinem**
Usage of the `File/Blob/Index` abstraction for `EventAttachment`s is problematic for various reasons.

The `FileBlob` deduplication is a bottleneck in attachment ingestion and has caused INC-513 among other things. The `File` abstraction also covers *every* kind of file we store, making it hard to distinguish different usecases like attachments and debug files. The `cleanup` of `EventAttachment` can barely keep up with writes, primarily due to the inefficiency of deleting the backing `File`.

This PR is doing the first preparatory changes to be able to move away from `File` to an alternative backing store. It duplicates some of the metadata that currently lives in `File` directly to `EventAttachment`. (Note that things like `name` and `type` were already duplicated) It also adds a new `blob_path` field and makes `file_id` nullable to be forward compatible to a future switch here.

As a driveby change, it also cleans up some of the indexes on the `EventAttachment` table that do not make sense.

### CODE REVIEW

**Review by loewenheim — APPROVED** [review]

File: src/sentry/migrations/0590_eventattachment_metadata.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        ),
+        migrations.AddField(
+            model_name="eventattachment",
+            name="sha1",
+            field=models.CharField(max_length=40, null=True),
+        ),
+        migrations.AddField(
+            model_name="eventattachment",
+            name="size",
+            field=sentry.db.models.fields.bounded.BoundedPositiveIntegerField(null=True),
```
Comment (markstory) [inline src/sentry/migrations/0590_eventattachment_metadata.py:45]: Should this be a bigint? Would we have files larger than 2,147,483,647 bytes?

Comment (Swatinem) [reply src/sentry/migrations/0590_eventattachment_metadata.py:45]: no, for two reasons:
- the existing `File` model has the same restriction and we do not support files over 2G.
- attachments specifically have even stricter size limits that are already enforced on the relay layer.

**Review by markstory — APPROVED** [review]
Schema changes look good to me.

**Comment by Swatinem** [comment]
According to @beezz , next steps here would be:

> The next thing would be to update the migration to not include the check constraint and add the check constraint in a separate migration is_dangerous = True so we can add it as invalid and validate async.

Either by adding the constraints at NOT VALID in when adding the column, or probably by just defining the field as `BoundedIntegerField` first, and then later changing that to a `BoundedPositiveIntegerField`.

**Pull request merged**